libs/ui/src/components/RecordEdition/hooks/useGetSubmitButtons.tsx

```diff
@@ -0,0 +1,51 @@
+import {KitButton} from 'aristid-ds';
+import {useSharedTranslation} from '_ui/hooks/useSharedTranslation';
+import {possibleSubmitButtons, submitButtonsProp} from '../_types';
+import {FontAwesomeIcon} from '@fortawesome/react-fontawesome';
+import {faPenToSquare, faCheck} from '@fortawesome/free-solid-svg-icons';
+import {EDIT_OR_CREATE_RECORD_FORM_ID} from '../EditRecordContent/formConstants';
+
+export const useGetSubmitButtons = (
+    buttons: submitButtonsProp,
```

Comment:
Je pense que cette props n'est pas forcement utile. On pourrait juste se baser sur le fait de passer `onCreate` et `onCreateAndEdit`. En fonction de si une prop est passée alors on affiche le bouton correspondant. Par exemple
```suggestion
    if (onCreate !== undefined) {
        headerSubmitButtons.push(
            <KitButton
                key="create"
                form={EDIT_OR_CREATE_RECORD_FORM_ID}
                type={buttons === 'create' ? 'primary' : 'secondary'}
                htmlType="submit"
                icon={<FontAwesomeIcon icon={faCheck} />}
                onClick={() => onClickSubmit('create')}
            >
                {t('record_edition.create')}
            </KitButton>
        );
    }
```
Ca permet aussi d'alléger les composants EditRecordPage et EditRecordModal avec une props de moins :)

Reply:
Je suis pas trop fan des props "magiques" qui font apparaître des éléments de manière implicite. En plus, on pourrait vouloir le bouton "créer et éditer" mais ne pas avoir de besoin d'une action particulière dans ce cas là (donc pas besoin de passer `onCreateAndEdit` )

Reply:
Non, la "redirection" vers le formulaire d'édition est gérée par le composant `EditRecordPage` ou `EditRecordModal`. La props `onCreateAndEdit` permet, si besoin, de faire une autre action après la création (rafraîchir un cache par exemple). Mais il peut y avoir des cas ou on a rien besoin de faire en plus.

Reply:
Je rejoins @TdyP sur les props "magiques" qui sont parfois déstabilisantes. De plus comme il dit : on peut avoir le bouton sans mettre de callback et laisser le comportement par défaut.

Reply:
ça roule je l'avais pas vu sous cet angle